guides/micronaut-cloud-secrets-azure/micronaut-cloud-secrets-azure.adoc

```diff
@@ -0,0 +1,194 @@
+      client-id: ${AZURE_CLIENT_ID} # <3>
+      tenant-id: ${AZURE_TENANT_ID} # <4>
+      secret: ${AZURE_SECRET_VALUE} # <5>
+  key-vault:
+    vault-url: ${AZURE_VAULT_URI} # <6>
+----
+<1> The application name entry is moved from `src/main/resources/application.yml` for resolving distributed configurations properties
+<2> Flag to enable the distributed configurations
+<3> The client ID of the trusted application that you will register in Azure Portal
+<4> The tenant ID
+<5> The secret value of the registered application
+<6> The Azure Key Vault resource URI
+
+NOTE: More details on how to obtain these values are provided in the Azure Portal section below.
+
+https://docs.micronaut.io/latest/guide/#bootstrap[Click here to get more details on distributed configuration]
+
+=== Clean up of the Application Configuration
+
+Remove the redundant application name entry from the `application.yml`.
+
+[,diff]
+.src/main/resources/application.yml
+----
+- micronaut:
+-  application:
+-    name: micronautguide
+
+datasources:
+  default:
+    username: ${JDBC-USER}
+    password: ${JDBC-PASSWORD}
+    dialect: MYSQL
+    driverClassName: ${JDBC-DRIVER:com.mysql.cj.jdbc.Driver}
+
+flyway:
+  datasources:
+    default:
+      enabled: true
+----
+
+== Store JDBC secrets in Azure Key Vault using Azure Portal
+
+Create Azure Key Vault service resource.
+{empty} +
+
+image::azure-key-vault/azure-vault-create-vault-resouce.png[]
+{empty} +
+
+When Key Vault service deployment is ready, go to the resource in order to create secrets. Prior to clicking on the Secrets option in the left menu list, copy the Vault URI from the main screen section. You will need to set it as an environment variable in order to integrate Azure Key Vault into your application.
+{empty} +
+
+image::azure-key-vault/azure-vault-obtain-vault-uri.png[]
+{empty} +
+
+Create new secrets for JDBC user and password. Click on the Secrets option in the left menu, then click on Generate/Import.
+{empty} +
+
+image::azure-key-vault/azure-vault-create-secret-jdbc-user.png[]
+{empty} +
+
+image::azure-key-vault/azure-vault-create-secret-jdbc-password.png[]
+{empty} +
+
+Go to App registration in order to register your application with the Microsoft identity platform. After registering your application,
+copy the secret Application (client) ID, and Directory (tenant) ID. You will use them later on as environment variables in order to connect to Azure Key Vault.
+{empty} +
+
+image::azure-key-vault/azure-register-application.png[]
+{empty} +
+
+Navigate to your application profile in order to generate application credentials that you will use to authenticate with the Azure Key Vault service. In the left menu, click on the Certificates & secrets. To generate the application credentials, click on the Client secrets tab and then on the New client secret button. Copy the secret value. You will need to set it along the Application (client) ID, Directory (tenant) ID, and Vault URI as environment variables.
```

Comment:
```suggestion
Navigate to your application profile to generate application credentials that you will use to authenticate with the Azure Key Vault service. In the left menu, click on the Certificates & secrets. To generate the application credentials, click on the Client secrets tab and then on the New client secret button. Copy the secret value. You must set it along the Application (client) ID, Directory (tenant) ID, and Vault URI as environment variables.
```

Reply:
Covered in https://github.com/micronaut-projects/micronaut-guides/pull/1099/commits/27ffa927a580938d2c8edd451d078f4c25369a2f.